src/main/webapp/app/entities/programming-exercise/programming-exercise-test-schedule-picker/programming-exercise-test-schedule-date-picker.component.ts

```diff
@@ -0,0 +1,90 @@
+import { Component, ElementRef, EventEmitter, forwardRef, Input, Output, ViewChild } from '@angular/core';
+import { isDate, Moment } from 'moment';
+import { ControlValueAccessor, NG_VALUE_ACCESSOR } from '@angular/forms';
+
+@Component({
+    selector: 'jhi-programming-exercise-test-schedule-date-picker',
+    template: `
+        <div>
+            <div>
+                <span class="font-weight-bold" [jhiTranslate]="label"></span>
+                <jhi-help-icon *ngIf="tooltipText" placement="top" [text]="tooltipText"></jhi-help-icon>
+            </div>
+            <div class="invisible-date-time-picker">
+                <input class="form-control" [ngModel]="val" [min]="min" [max]="max" (ngModelChange)="updateField($event)" [owlDateTime]="dt" />
+            </div>
+            <button *ngIf="!val" [owlDateTimeTrigger]="dt" type="button" class="btn btn-light btn-lifecycle" [class.btn-lifecycle--deactivated]="!val">
+                <fa-icon class="icon-calendar-plus" icon="calendar-plus" size="2x"></fa-icon>
+            </button>
+            <button *ngIf="val" (click)="resetDate()" type="button" class="btn btn-light btn-lifecycle calendar-event-toggle">
+                <fa-icon class="icon-static" icon="calendar-check" size="2x"></fa-icon>
+                <fa-icon class="icon-remove" icon="calendar-minus" size="2x"></fa-icon>
+            </button>
+            <div *ngIf="val">
+                {{ val | date: 'MMM, dd' }}<br />
+                {{ val | date: 'HH:mm' }}
+            </div>
+            <div *ngIf="!val" class="text-danger" jhiTranslate="artemisApp.programmingExercise.timeline.notSet">
+                not set
+            </div>
+            <owl-date-time [startAt]="startAt" #dt></owl-date-time>
+        </div>
```

Comment:
Same as above, I think this is enough to justify its own file.

Reply:
I also moved this